src/app/shell/details/details-tabs/all-provider-workshops/all-provider-workshops.component.html

```diff
@@ -1,7 +1,22 @@
-<div class="card-wrapper" *ngIf="workshops?.length; else isEmptyList">
-  <app-workshop-card class="card" *ngFor="let workshop of workshops" [workshop]="workshop"></app-workshop-card>
-</div>
+<ng-container *ngIf="!!workshops">
+  <div class="card-wrapper" *ngIf="workshops?.totalAmount; else isEmptyList">
+    <app-workshop-card
+      class="card"
+      *ngFor="let workshop of workshops.entities"
```

Comment:
Please provide trackBy function

Reply:
[NgForOf](https://angular.io/api/common/NgForOf) needs to uniquely identify items in the iterable to correctly perform DOM updates when items in the iterable are reordered, new items are added, or existing items are removed.

but in our case we have changes from the server and it is always a new list